Makefile

```diff
@@ -325,7 +341,24 @@ kustomize-crds: output.init $(KUSTOMIZE) $(YQ)
 	XDG_CONFIG_HOME=$(PWD)/package $(KUSTOMIZE) build --enable-alpha-plugins $(OUTPUT_DIR)/package/kustomize -o $(OUTPUT_DIR)/package/crds.yaml || $(FAIL)
 	@$(OK) Kustomizing CRDs.
 
+.PHONY: kustomize-crds
+
 checkout-to-old-api:
 	CHECKOUT_RELEASE_VERSION=$(CHECKOUT_RELEASE_VERSION) hack/check-duplicate.sh
 
-.PHONY: kustomize-crds
+ifeq ($(RUN_BUILDTAGGER),true)
+lint.init: build-lint-cache
+lint.done: delete-build-tags
+
+build-lint-cache: $(GOLANGCILINT)
+	@$(INFO) Running golangci-lint with the analysis cache building phase.
+	@(BUILDTAGGER_DOWNLOAD_URL=$(BUILDTAGGER_DOWNLOAD_URL) ./scripts/tag.sh && \
+	(([[ "${SKIP_LINTER_ANALYSIS}" == "true" ]] && $(OK) "Skipping analysis cache build phase because it's already been populated") && \
+	[[ "${SKIP_LINTER_ANALYSIS}" == "true" ]] || $(GOLANGCILINT) run -v --build-tags account,configregistry,configprovider,linter_run -v --concurrency 1 --disable-all --exclude '.*')) || $(FAIL)
```

Comment:
nit: Can we consider increasing the Concurrency?

Reply:
A concurrency of 4 for the initial analysis cache build phase has failed here:
https://github.com/upbound/provider-aws/actions/runs/8246986803/job/22554209938?pr=1194

Reply:
[Here](https://github.com/upbound/provider-aws/actions/runs/8235027938/job/22518207849)'s a run with a concurrency of 1. It took ~21 min and ~14 GB of peak memory to complete the initial phase:
<img width="592" alt="image" src="https://github.com/upbound/provider-aws/assets/9376684/c01c71b3-c094-4fcd-a07e-76eb04b73016">

And [here](https://github.com/upbound/provider-aws/actions/runs/8247646010/job/22556266352?pr=1194)'s another run with a concurrency of 3. It took ~14 min ~19 GB of peak memory to complete the initial phase:
<img width="581" alt="image" src="https://github.com/upbound/provider-aws/assets/9376684/0befe5d2-0b13-4d62-a8b2-95dd58ead668">

Let's switch to a concurrency of 3 for the initial phase (concurrency of 4 has failed). The second phase already uses the max available concurrency. 

If we observe any memory issues for the initial phase, we can decrease it later again.